src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -413,6 +449,96 @@ def read_file(
                 result_df = pd.concat([result_df, df], ignore_index=True)
             result_df.sf_types = result_df_sf_types
             return result_df
+        elif file_format == "json":
+            infer_schema_opt = options.get("INFER_SCHEMA", False)
+
+            result_df = TableEmulator()
+            result_df_sf_types = {}
+
+            if not infer_schema_opt:
+                # if infer schema option is False, then snowflake converts the data into
+                # a single column table, values are treated as raw strings
+                assert len(schema) == 1, (
+                    f"[Local Testing] Unexpected schema length {len(schema)} when loading "
+                    f"json data without inferring schema."
+                )
+                column_name = analyzer.analyze(schema[0])
+                column_series = ColumnEmulator(
+                    data=None,
+                    dtype=object,
+                    name=column_name,
+                    sf_type=ColumnType(VariantType(), True),
+                )
+                result_df[column_name], result_df_sf_types[column_name] = (
+                    column_series,
+                    column_series.sf_type,
+                )
+
+                for local_file in local_files:
+                    with open(local_file) as file:
+                        content = json.load(file, cls=CUSTOM_JSON_DECODER)
+                        df = pd.DataFrame({result_df.columns[0]: [content]})
+                        result_df = pd.concat([result_df, df], ignore_index=True)
+            else:
+                # need to infer schema
+                contents = []
+                for local_file in local_files:
+                    with open(local_file) as file:
+                        content = json.load(file, cls=CUSTOM_JSON_DECODER)
+                        tmp_content = {}
+                        # snowflake escape double quotes by adding extra double quote
+                        for key, value in content.items():
```

Comment:
This makes sense 👍 , can we reuse https://github.com/snowflakedb/snowpark-python/blob/bf4a409fb78816568de69e57da6e2bf6919643d7/src/snowflake/snowpark/_internal/utils.py#L855?